### PR TITLE
Unified mbed copy script

### PR DESCRIPTION
**Pull request by justbuchanan**
I tweaked your osx mbed-copy script a bit so that it works on linux too.  The main difference is that it will automatically mount mbeds that aren't mounted.  Unfortunately this requires sudo, so I added sudo to the calls of the script.  I also changed `requirements2.txt` to point to my fork of `mbed-ls`, which includes your OS X changes and my Linux changes.

### CODE REVIEW

**Comment by justbuchanan** [comment]
@jjones646, did you get a chance to test this out and make sure it works on your computer?

**Comment by jjones646** [comment]
@justbuchanan Ahh, didn't see this...let's give it a go.

It doesn't like rebooting when screened into the mbed. Despite the screen session having regular user privileges and that script running with sudo.
This is on OS X - Ubuntu seems to work fine. Running python 2.7.11 on OS X and python 2.7.6 on Ubuntu. Same version of pyserial in use, version 3.0.1.

```
[100%] Copying the robot's binary over to the mbed
Copying 'rj-robot.bin' to '/Volumes/MBED'...
Rebooting mbed...
Traceback (most recent call last):
  File "/Users/jonathan/Documents/robocup-software-watson/util/mbed/mbed-copy.py", line 85, in <module>
    ss = serial.Serial(mbed['serial_port'], baudrate=57600)
  File "/usr/local/lib/python2.7/site-packages/serial/serialutil.py", line 180, in __init__
    self.open()
  File "/usr/local/lib/python2.7/site-packages/serial/serialposix.py", line 294, in open
    raise SerialException(msg.errno, "could not open port %s: %s" % (self._port, msg))
serial.serialutil.SerialException: [Errno 16] could not open port /dev/tty.usbmodem1422: [Errno 16] Resource busy: '/dev/tty.usbmodem1422'
make[4]: *** [firmware/robot2015/src-ctrl/CMakeFiles/robot2015-prog] Error 1
make[3]: *** [firmware/robot2015/src-ctrl/CMakeFiles/robot2015-prog.dir/all] Error 2
make[2]: *** [firmware/robot2015/src-ctrl/CMakeFiles/robot2015-prog.dir/rule] Error 2
make[1]: *** [robot2015-prog] Error 2
make: *** [robot2015-prog] Error 2
```

Somewhat irrelevant, but it is worth noting that I commented out the comm thread from starting up for all of the above. The mbed crashes with this branch otherwise (so no serial console unless I do that). For completeness, here's the diff.

```
-    Thread comm_task(Task_CommCtrl, mainID, osPriorityAboveNormal);
-    Thread::signal_wait(MAIN_TASK_CONTINUE, osWaitForever);
+    // Thread comm_task(Task_CommCtrl, mainID, osPriorityAboveNormal);
+    // Thread::signal_wait(MAIN_TASK_CONTINUE, osWaitForever);

-    comm_task.signal_set(SUB_TASK_CONTINUE);
+    // comm_task.signal_set(SUB_TASK_CONTINUE);
```

Other than that minor error, it seems to work good.

File: firmware/CMakeLists.txt [hunk 1]
```diff
@@ -11,14 +11,7 @@ remove_cxx_flag("-fcolor-diagnostics")
...
 # set the path to a script for moving files to the mbed's USB storage
-if( UNIX AND NOT APPLE )
-    set(MBED_COPY_SCRIPT "${PROJECT_SOURCE_DIR}/util/mbed/mbed-copy.sh")
-elseif( APPLE )
-    set(MBED_COPY_SCRIPT "${PROJECT_SOURCE_DIR}/util/mbed/mbed-copy-osx.py")
-else()
-    message(WARNING "unsupported host system for mbed-copy usage")
-    set(MBED_COPY_SCRIPT "")
-endif()
+set(MBED_COPY_SCRIPT "${PROJECT_SOURCE_DIR}/util/mbed/mbed-copy.py")
```
Comment (jjones646) [inline firmware/CMakeLists.txt]: Here's a cleaner recommendation for sudo that should work I believe.

```
set(MBED_COPY_SCRIPT "sudo ${PROJECT_SOURCE_DIR}/util/mbed/mbed-copy.py")
```

**Comment by jjones646** [comment]
@justbuchanan The serial exception could be related to ubuntu using an alias to screen into the mbed.

File: firmware/CMakeLists.txt (hunk 1 above)
Comment (justbuchanan) [inline firmware/CMakeLists.txt]: :+1:, just changed it

**Comment by justbuchanan** [comment]
Interesting, I'll look into it.  I don't have my Mac with me at the moment, but I'll poke around with this when I get back home later tonight.

**Comment by justbuchanan** [comment]
I tried it out on my Mac and I'm getting the same error you got.  It seems like Linux allows multiple programs to connect to the port simultaneously, but OS X limits it to one connection.  I tried out the version on your branch and it hits the same error, it just doesn't print it out because it's wrapped in a try/catch block.

I googled around a little and I don't see anything we can do to get around this.  Maybe the best option would be to detect the exception if it happens and just print a helpful message suggesting that it might not be working because screen is connected to the serial port.

**Comment by jjones646** [comment]
I've ran across a similar issue with pyusb before. All I remember is that it became incredibly complicated really quickly. Something about having to release the kernel driver, use it, then reset the kernel driver. See [here](http://www.resolvinghere.com/sof/8218683.shtml). Most likely not worth the trouble.

**Pull request merged**